### ISSUE TITLE
stat: missing operand AND SQLite errors - new server - new TS server install - script 5.6.6

### ISSUE THREAD

**Issue by sanjiyan** [TS3Tools/TS3UpdateScript#93]
Hi all,

Due to the problems of my previous TS server install, I have blown it away and the Ubuntu server, and redeployed.

Happy to say it is working fine, untill I ran TS3Info update script to verify that everything works including the update script.

Server: Ubuntu 18.04.1 LTS
TS Server 3.4.0
TSUpdateScript version: 5.6.6 

I get the following errors:

`

> All collected information were validated successfully as preventive measures against possible failures                                                                                                                                                                                                             [ INFO ]
> Try 'stat --help' for more information.
> 1 TeamSpeak 3 server instance found...or directory
> stat: missing operand
> Collecting as much as possible information for following TeamSpeak 3 server instance: /usr/local/teamspeak
> 5 instances for update process...file or directory
> /usr/local/teamspeak: This instance is already up2date. Nothing to do.
> ./TS3UpdateScript: line 1036: cd: SQLite/SQLite/logs/: No such file or directory
> ./TS3UpdateScript: line 1036: cd: SQLite/SQLite/logs/: No such file or directory
> stat: missing operand
> Try 'stat --help' for more information.
> -su: line 0: cd: SQLite: No such file or directory
> 
> Following instance needs an update, but is stopped and not touched by the script: SQLite (Installed Version: SQLite)
> 
> Cleaning up...
> 
> Summary of measured execution time for each step
> Total script: 7s
> Self-Test: 4s
> Collecting Information: 2s
> Validation: 1s
> Execution Mechanism: 0s
> Cleanup: 0s

`

Now the TS server is 3.4.0 and it is still running (as shown below)

`

> service teamspeak status
> ● teamspeak.service - LSB: TeamSpeak Server Daemon
>    Loaded: loaded (/etc/init.d/teamspeak; generated)
>    Active: active (exited) since Fri 2018-09-28 07:46:09 EDT; 5 days ago
>      Docs: man:systemd-sysv-generator(8)
>     Tasks: 0 (limit: 1110)
>    CGroup: /system.slice/teamspeak.service
> 
> Sep 28 07:46:08 ts.sanjiyan.com systemd[1]: Starting LSB: TeamSpeak Server Daemon...
> Sep 28 07:46:09 ts.sanjiyan.com su[781]: Successful su for teamspeak by root
> Sep 28 07:46:09 ts.sanjiyan.com su[781]: + ??? root:teamspeak
> Sep 28 07:46:09 ts.sanjiyan.com su[781]: pam_unix(su:session): session opened for user teamspeak by (uid=0)
> Sep 28 07:46:09 ts.sanjiyan.com teamspeak[777]: Starting the TeamSpeak 3 server
> Sep 28 07:46:09 ts.sanjiyan.com teamspeak[777]: TeamSpeak 3 server started, for details please view the log file
> Sep 28 07:46:09 ts.sanjiyan.com su[781]: pam_unix(su:session): session closed for user teamspeak
> Sep 28 07:46:09 ts.sanjiyan.com systemd[1]: Started LSB: TeamSpeak Server Daemon.

`

The TS server is installed in /usr/local/teamspeak
Files and directorys are owned by user:group: teamspeak:teampeak
Service is ran under user: teamspeak

` 845 teamspe+  20   0  695756  23772  11244 S  1.0  2.4  32:16.37 ts3server`

I have as far as I know, installed all the required and optional files, applications, programs via apt-get.

I have attached the error.txt file.

Does anyone have any ideas?

Regards.
[error.txt](https://github.com/TS3Tools/TS3UpdateScript/files/2444279/error.txt)

**Comment by Sebi94nbg** [comment]
Hi @sanjiyan ,
Thanks for creating this issue.

Have you installed the packages `telnet` and `expect`? It seems like as you don't have installed them and due to this, the script tries to gather all information from the latest instance log file `_0.log`.

In your case, it finds all values multiple times and have an issue due to this as it expects each value only once in each log file. This issue usually happens, when you append the log after a restart of the instance instead of deleting the log file or creating a new one (default behaviour).

Using `telnet` and `expect`, the script can detect much better all necessary information as it's fetching those using the ServerQuery commands.

If you don't use those packages, please try to solve this issue by stopping the instance, deleting all log files and starting the instance. After this, the script should work fine.

In the meanwhile, I'll try to reproduce this issue in order to fix it, if possible.

**Comment by sanjiyan** [comment]
Hi there,

I can confirm that telnet and expect are installed:
```

root@ts:~# apt-get install telnet expect
Reading package lists... Done
Building dependency tree
Reading state information... Done
telnet is already the newest version (0.17-41).
expect is already the newest version (5.45.4-1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

**Comment by sanjiyan** [comment]
Happy to confirm that changing the ts3server.ini file 'logappend=1' to 'logappend=0' fixed the errors.

But I would prefer to use telnet and expect, rather than rely on log files.

So I would like to continue with this ticket, please.

**Comment by Sebi94nbg** [comment]
Detection with enabled `appendlog` has been fixed with release [5.8.0](https://github.com/TS3Tools/TS3UpdateScript/releases/tag/5.8.0).
